Bottom: d9c397bbaebf677dc1babf06221f3e81ba0f7c8f
Top:    b2d529be1a7b43c622d3c288dc0de67df0bfb421
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-02-17 19:43:58 +0800

Refresh of virtio-net-small-buffer-xdp

---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 7e834b3..765b65e 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -1997,10 +1997,7 @@ static void free_unused_bufs(struct virtnet_info *vi)
 	for (i = 0; i < vi->max_queue_pairs; i++) {
 		struct virtqueue *vq = vi->sq[i].vq;
 		while ((buf = virtqueue_detach_unused_buf(vq)) != NULL) {
-			if (!is_xdp_raw_buffer_queue(vi, i))
-				dev_kfree_skb(buf);
-			else
-				put_page(virt_to_head_page(buf));
+			put_page(virt_to_head_page(buf));
 		}
 	}
